oepcie/yuanphil_impl/oe_ctx.h

```diff
@@ -0,0 +1,31 @@
+#ifndef _INCL_GUARD_ONE
+#define _INCL_GUARD_ONE
+
+#include "oe_ctx_impl.h"
+
+typedef struct oe_ctx_impl *oe_ctx;
+
+int oe_init(oe_ctx* ctx);
+
+oe_ctx* oe_create();
+
+void oe_destroy(oe_ctx* ctx);
+
+int oe_set_opt(oe_ctx* state, int option, const char* value);
+
+/*
+ * For functional purposes, here we are going to assume that value points to
+ * NULL, and we will allocate the string to return. It is up to the use to cl-
+ * ean up memory.
+ */
+int oe_get_opt(oe_ctx* state, int option, char* value);
+
+int oe_write_reg(oe_ctx* state, int device_id, int addr, int value);
```

Comment:
If the lseek and write calls may modify the file descriptors, the "ctx" in oe_read_reg() also can't be const since it has to send the read_trigger to the hardware.